### DISCUSSION TITLE
Missions with "arbitrary" location of key/value (channelized) pairs to represent mnemonics in the CCSDS user data field

### DISCUSSION THREAD

**Discussion by haisamido** [CCSDSPy/ccsdspy#116] [Ideas]
@ddasilva There are some missions where mnemonics are packed into the CCSDS Space Packet's User Data Field (UDF) in a near arbitrary bit location, where the beginning of the bit location is the beginning of the "key" with a specific bit size, and what immediately follows is the actual value of the mnemonic. Sometimes these types of mnemonics are known as channelized mnemonics. This allows for a very small packet for low bandwidth missions. The consequence of this is that the key has to be searched for in the user data field, and its location can be dynamically placed in UDF. Are there plans to support such a Space Packet?

**Comment by ddasilva** [comment]
Hi @haisamido, we've talked about this feature.. but currently there are no plans to build this in. 

You can currently scan for the pattern yourself and pass the relevant bytes you find into the load() function of a packet as BytesIO. There are also functions in the utils module to split a file into a list of bytes objects by packet, should that be useful.
